pkg/capabilities/consensus/ocr3/requests/handler.go

```diff
@@ -0,0 +1,167 @@
+package requests
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+type responseCacheEntry struct {
+	response  *Response
+	entryTime time.Time
+}
+
+type Handler struct {
+	services.StateMachine
+
+	lggr logger.Logger
+
+	store *Store
+
+	pendingRequests map[string]*Request
+
+	responseCache   map[string]*responseCacheEntry
+	cacheExpiryTime time.Duration
+
+	responseCh chan *Response
+	requestCh  chan *Request
+
+	clock clockwork.Clock
+
+	stopCh services.StopChan
+	wg     sync.WaitGroup
+}
+
+func NewHandler(lggr logger.Logger, s *Store, clock clockwork.Clock, responseExpiryTime time.Duration) *Handler {
+	return &Handler{
+		lggr:            lggr,
+		store:           s,
+		pendingRequests: map[string]*Request{},
+		responseCache:   map[string]*responseCacheEntry{},
+		responseCh:      make(chan *Response),
+		requestCh:       make(chan *Request),
```

Comment:
Is there a risk that Handler will get closed first and then the Capability will hang forever on SendRequest()?

Reply:
true, it shoudl pass in the cts and select on that, sorting......